sql/src/main/java/io/crate/metadata/IngestionRuleInfos.java

```diff
@@ -0,0 +1,37 @@
+package io.crate.metadata;
+
+
+import io.crate.metadata.rule.ingest.IngestRulesMetaData;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import static java.util.Collections.emptyIterator;
+
+public class IngestionRuleInfos implements Iterable<IngestionRuleInfo> {
+
+    private List<IngestionRuleInfo> ingestionRuleInfoList = null;
+
+    public IngestionRuleInfos(IngestRulesMetaData ingestRulesMetaData) {
+        if (ingestRulesMetaData != null && ingestRulesMetaData.getIngestRules() != null) {
+            ingestionRuleInfoList = new ArrayList<>();
+            ingestRulesMetaData.getIngestRules().forEach((key, value) -> value.forEach(ingestRule ->
+                ingestionRuleInfoList.add(new IngestionRuleInfo(
+                    ingestRule.getName(),
+                    key,
+                    ingestRule.getTargetTable(),
+                    ingestRule.getCondition()))
+            ));
+        }
+    }
+
+    @Override
+    public Iterator<IngestionRuleInfo> iterator() {
+        if (ingestionRuleInfoList == null) {
```

Comment:
Could make the list final and assign to an empty list in the constructor to avoid this check

Reply:
Wouldn't that simply convert the check from "== null" to "isEmpty" ? Assuming we still want to return the singleton `emptyIterator`

Reply:
`emptyList().iterator()` returns `emptyIterator()` - it's always the same instance, so we don't have to do any check here  - and in the constructor there is already a if branch anyway